test/common/tmpdir.js

```diff
@@ -1,31 +1,56 @@
 /* eslint-disable node-core/require-common-first, node-core/required-modules */
 'use strict';
 
+const { execSync } = require('child_process');
 const fs = require('fs');
 const path = require('path');
+const { debuglog } = require('util');
 
-function rimrafSync(p) {
-  let st;
-  try {
-    st = fs.lstatSync(p);
-  } catch (e) {
-    if (e.code === 'ENOENT')
-      return;
+const debug = debuglog('test/tmpdir');
+
+function rimrafSync(pathname) {
+  const st = (() => {
+    try {
+      return fs.lstatSync(pathname);
+    } catch (e) {
+      if (fs.existsSync(pathname))
+        throw new Error(`Something wonky happened rimrafing ${pathname}`);
+      debug(e);
+    }
+  })();
+
+  // If (!st) then nothing to do.
+  if (!st) {
+    return;
+  }
+
+  // On Windows first try to delegate rmdir to a shell.
+  if (process.platform === 'win32' && st.isDirectory()) {
+    try {
+      execSync(`rmdir /q /s ${pathname}`, { timout: 1000 });
```

Comment:
If everything passes, should this not return?

Reply:
Not totally sure why, but it's not trivial https://ci.nodejs.org/job/node-test-binary-windows-2/1275/